lib/src/asset_graph/exceptions.dart

```diff
@@ -11,3 +11,14 @@ class DuplicateAssetNodeException implements Exception {
   @override
   String toString() => 'DuplicateAssetNodeError: $assetNode';
 }
+
+class AssetGraphVersionError implements Exception {
+  final int versionSeen;
+  final int currentVersion;
+
+  AssetGraphVersionError(this.versionSeen, this.currentVersion);
+
+  @override
+  String toString() => 'AssetGraphVersionError: saw version $versionSeen but '
```

Comment:
Is it possible to be more concrete? What component of the system saw (read? deserialized?) the incorrect version?